binder/src/main/java/io/grpc/binder/internal/ActiveTransportTracker.java

```diff
@@ -0,0 +1,114 @@
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.HashSet;
+import javax.annotation.concurrent.GuardedBy;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransportListener;
+
+/**
+ * Tracks which {@link BinderTransport.BinderServerTransport} are currently active and allows
+ * invoking a {@link BinderTransportSecurity.TerminationListener} only once all transports are
+ * terminated.
+ */
+final class ActiveTransportTracker {
+  @GuardedBy("this")
+  private final HashSet<BinderTransport.BinderServerTransport> activeTransports = new HashSet<>();
+  private final BinderTransportSecurity.TerminationListener transportSecurityTerminationListener;
+  @GuardedBy("this")
+  private boolean awaitingNoActiveTransports = false;
+  @GuardedBy("this")
+  private boolean terminationNotificationRequested = false;
```

Comment:
Can you call this `shutdown` instead? I think it would be a lot clearer.

Reply:
Updated.